test/replication-luatest/election_fencing_test.lua

```diff
@@ -0,0 +1,293 @@
+    box_cfg_update({g.server_2}, {replication = {}})
+    wait_disconnected(g.server_1, g.server_2)
+    local ok = test_rw(g.server_1)
+    luatest.assert(ok, 'Leader is rw after quorum loss')
+
+    -- Turning on fencing on leader when quorum is allready lost must make it ro
+    box_cfg_update({g.server_1}, {election_fencing_enabled = true})
+    local ok, err = test_rw(g.server_1)
+    luatest.assert(not ok and err.code == box.error.READONLY,
+        'Leader is ro after quorum loss')
+end
+
+g_sync.before_all(start)
+g_sync.after_all(stop)
+
+g_sync.test_fencing = function(g)
+    box_cfg_update(g.cluster.servers, {read_only = false})
+    g.server_1:exec(function()
+        box.schema.create_space('test', {is_sync = true}):create_index('pk')
+        box.cfg{election_mode = 'candidate'}
+    end)
+
+    box_cfg_update({g.server_1}, {
+        election_fencing_enabled = false,
+        replication_synchro_timeout = LONG_TIMEOUT,
+    })
+
+    box_cfg_update({g.server_2, g.server_3}, {replication = {}})
+    wait_disconnected(g.server_1, g.server_2)
+    wait_disconnected(g.server_1, g.server_3)
+
+    local leader, limbo_len = g.server_1:exec(function(t)
+        require('fiber').sleep(t)
+        return box.info.election.leader, box.info.synchro.queue.len
+    end, {DEATH_TIMEOUT})
+    luatest.assert_equals(leader, 1)
+    luatest.assert_equals(limbo_len, 0)
+
+    g.server_1:exec(function()
+        require('fiber').create(function() box.space.test:replace{1} end)
+    end)
+
+    -- Enabling fencing leads to leader resign.
+    box_cfg_update({g.server_1}, {election_fencing_enabled = true})
+    box_cfg_update({g.server_1}, {replication_synchro_timeout = SHORT_TIMEOUT})
+
+    -- Fenced leader must not CONFIRM/ROLLBACK unfinished synchronous
+    -- transactions.
+    local leader, limbo_len = g.server_1:exec(function(t)
+        require('fiber').sleep(t)
+        return box.info.election.leader, box.info.synchro.queue.len
+    end, {DEATH_TIMEOUT})
+    luatest.assert_equals(leader, 0)
+    luatest.assert_equals(limbo_len, 1)
+
+    -- After regaining quorum and becoming leader once again old leader must
+    -- replicate previously "frozen" synchronous transactions and confirm them.
+    box_cfg_update({g.server_2}, {replication = g.box_cfg.replication})
+    wait_connected(g.server_1, g.server_2)
+    g.server_1:wait_election_leader()
+    wait_sync(g.server_1, {g.server_2})
+    luatest.helpers.retrying({}, function()
+        luatest.assert(g.server_1:exec(function()
+            return box.info.synchro.queue.len == 0
+        end))
+    end)
+
+    local ret = g.server_1:exec(function()
+        return box.space.test:select{1}
+    end)
+    luatest.assert_equals(ret, {{1}},
+        'Sync write confirmed after leadership regain')
+
+    box_cfg_update({g.server_1}, {
+        election_fencing_enabled = false,
+        replication_synchro_timeout = LONG_TIMEOUT,
+    })
+
+    box_cfg_update({g.server_1}, {replication = {
```

Comment:
`{replication = ""}` would be sufficient.

Reply:
Fixed